src/action/listdir.rs

```diff
@@ -66,12 +66,11 @@ impl ParseError {
 
     /// Constructs an `UnsupportedValue` from provided `field` and
     /// `value`, then  converts it to `ParseError`.
-    fn unsupported_value(field: String, value: String) -> session::ParseError {
-        session::ParseError::malformed(ParseError::UnsupportedValue(
-            UnsupportedValueError {
-                field,
-                value,
-            }))
+    fn unsupported_value(field: String, value: String) -> ParseError {
+        ParseError::UnsupportedValue(UnsupportedValueError {
+            field,
+            value,
+        })
```

Comment:
You can make this function more general (which will simplify the code below), if instead of taking `String`, you take objects of type `Into<String>`. This way strings can still be passed into your function (because `Into<T>` is implemented for every `T` itself), but also things like string literals. The signature would like like:

```
fn unuspported_value<F, V>(field: F, value: V) -> ParseError
where
    F: Into<String>,
    V: Into<String>,
{ ... }
```